dolphinscheduler-task-plugin/dolphinscheduler-task-procedure/src/main/java/org/apache/dolphinscheduler/plugin/task/procedure/ProcedureParameters.java

```diff
@@ -104,6 +104,14 @@ public void dealOutParam4Procedure(Object result, String pop) {
             return;
         }
         properties.get(pop).setValue(String.valueOf(result));
+        if (!varPool.isEmpty()) {
+            for (Property info : varPool) {
+                if (pop.equals(info.getProp())) {
+                    varPool.remove(info);
+                    break;
+                }
+            }
+        }
```

Comment:
You should fix this in `AbstractParameters` since this is used by more than just one task type.

Reply:
other task type use the mothod of AbstractParameters.dealOutParam .only procedure task handle output parameters in this way.

Reply:
What I mean is remove duplicated value of `varPool` in `AbstractParameters`.

Reply:
> What I mean is remove duplicated value of `varPool` in `AbstractParameters`.

![图片](https://github.com/user-attachments/assets/36cefe83-2e8b-4f45-a902-db94b5541277)
this is the method merge the out params in AbstractParameters, the inputparams of this method is varpool and outProperty,This is sequential. It will be inserted into the map according to the order of varpool and then outProperty. The map will filter out duplicate values, and because outProperty is behind, it will overwrite the duplicate values of varpool.Do you mean this？

Reply:
Yes.

Reply:
So the mergeVarPool of AbstractParameters does not need to be modified

Reply:
After I take a deep look at it, I think we should make the following changes.
1. Deduplication should be performed at the place where varpool was initialized.
2. Add duplicate check at the front-end.
3. Add duplicate check at the backend-end api.

Reply:
I think parameter updates should be a normal business scenario